### DISCUSSION TITLE
tail -f command is not working as I expected

### DISCUSSION THREAD

**Discussion by deterset** [microsoft/WSL#8786] [Q&A] [answered]
When using tail -f results are not displayed as text get added to the file your monitoring.

I have found that adding ---disable-inotify on the command line fixes it. My memory is not the greatest so I added an alias to my .bashrc file near all of the aliases. Here is an example.

alias tail='tail ---disable-inotify'

Now I can use tail -f and it works as expected, or as I expect it to.
Please let me know if there are any issues always having ---disable-inotify added to the command on a WSL Linux install.

I just wanted to share my success.

Thanks,
Robert

**Comment by NotTheDr01ds** [comment, marked as answer]
Good find on the `---disable-inotify` option.  That's one I hadn't seen.  It's not even in the manpage -- Apparently it is "deliberately undocumented" according to Jim Meyering, one of the `tail` contributors/maintainers in [this mail thread](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=583198#20).

> Please let me know if there are any issues always having

Not really an "issue", but just a note that you should only need `--disable-inotify` when trying to `tail -f` files on a *Windows* drive (e.g. `/mnt/c/...`.  Things should work normally on files *inside* WSL (e.g. `/var`, `/home`, etc.).

Disabling `inotify` support even on filesystems that *do* support it will simply mean that `tail` resorts to polling every (by default) 1 second.  With `inotify` support, on the other hand, you'll see updates as they occur rather than having to wait for an average 0.5 seconds.  Obviously, that shouldn't be a a huge deal.

And while I should mention that `sudo tail` won't respect the alias, by default, it probably doesn't matter, as you shouldn't ever need to `sudo tail -f` a file on a Windows drive anyway (your regular user should have permissions).

In case, for some reason, you do need to `sudo tail -f` on a Windows drive, you can also add `alias sudo='sudo '` as per [this Ask Ubuntu answer](https://askubuntu.com/a/22043/1165986).
